File: examples/fundamentals-java/src/main/java/example/integeroverflow/Numbers.java
package example.integeroverflow;

/** Two tiny routines: one that intuition gets wrong, one it gets right. */
public final class Numbers {

    private Numbers() {
    }

    /** Absolute value. BUG: {@code abs(Integer.MIN_VALUE)} overflows to a negative. */
    public static int abs(int x) {
        return x < 0 ? -x : x;
    }

    /** The larger of two values. Correct for all inputs. */
    public static int max(int a, int b) {
        return a >= b ? a : b;
    }
}

File: examples/fundamentals-kotlin/src/main/kotlin/example/integeroverflow/Numbers.kt
package example.integeroverflow

/** Two tiny routines: one that intuition gets wrong, one it gets right. */
object Numbers {

    /** Absolute value. BUG: `abs(Int.MIN_VALUE)` overflows back to a negative. */
    fun abs(x: Int): Int = if (x < 0) -x else x

    /** The larger of two values. Correct for all inputs. */
    fun max(a: Int, b: Int): Int = if (a >= b) a else b
}
